### PR TITLE
document GCS authentication

### PR DESCRIPTION
**Pull request by petedannemann**
#### document GCS authentication

Please **pick a concise, informative and complete title** for your PR.
The title is important because it will appear in [our change log](https://github.com/RaRe-Technologies/smart_open/blob/master/CHANGELOG.md).

#### Motivation

Please explain the motivation behind this PR in the description.

If you're fixing a bug, link to the issue number like so: 

- Add missing docs based on discussion [here](https://github.com/RaRe-Technologies/smart_open/pull/404#issuecomment-579240867)

If you're adding a new feature, then consider opening a ticket and discussing it with the maintainers before you actually do the hard work.

#### Tests

If you're fixing a bug, consider [test-driven development](https://en.wikipedia.org/wiki/Test-driven_development):

1. Create a unit test that demonstrates the bug. The test should **fail**.
2. Implement your bug fix.
3. The test you created should now **pass**.

If you're implementing a new feature, include unit tests for it.

Make sure all existing unit tests pass.
You can run them locally using:

    pytest smart_open

If there are any failures, please fix them before creating the PR (or mark it as WIP, see below).

#### Work in progress

If you're still working on your PR, include "WIP" in the title.
We'll skip reviewing it for the time being.
Once you're ready to review, remove the "WIP" from the title, and ping one of the maintainers (e.g. mpenkov).

#### Checklist

Before you create the PR, please make sure you have:

- [x] Picked a concise, informative and complete title
- [x] Clearly explained the motivation behind the PR
- [x] Linked to any existing issues that your PR will be solving
- [ ] Included tests for any new functionality
- [X] Checked that all unit tests pass

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -313,6 +313,36 @@ The ``version_id`` transport parameter enables you to get the desired version of
...
+
+To override this behavior, pass a ``google.cloud.storage.Client`` object as a transport parameter to the ``open`` function.
+You can `customize the credentials <https://google-cloud-python.readthedocs.io/en/0.32.0/core/client.html>`__
+when constructing the client. ``smart_open`` will then use the client when talking to GCS.
+
+.. code-block:: python
+
+    >>> from google.cloud.storage import Client
+    >>> client = Client.from_service_account_json('/path/to/keyfile.json')
+    >>> fin = open('gs://bucket/key', transport_params=dict(client=client), ...)
```
Comment (mpenkov) [inline README.rst]: Same thing with the URL. Ideally, it should be real. Is it difficult to set up a publicly readable GCS bucket and key?

File: README.rst [hunk 2]
```diff
@@ -313,6 +313,36 @@ The ``version_id`` transport parameter enables you to get the desired version of
...
+and pass it to the Client.
+
+.. code-block:: python
+
+	>>> import os
+	>>> from google.auth.credentials import Credentials
+	>>> token = os.environ['GOOGLE_API_TOKEN']
+	>>> credentials = Credentials(token=token)
+	>>> client = Client(credentials=credentials)
+	>>> fin = open('gs://bucket/key', transport_params=dict(client=client), ...)
```
Comment (mpenkov) [inline README.rst]: Please get rid of the ellipsis. These are real, executable examples, so we should not use the ellipsis in them.

File: README.rst [hunk 3]
```diff
@@ -313,6 +313,36 @@ The ``version_id`` transport parameter enables you to get the desired version of
...
+    >>> client = Client.from_service_account_json('/path/to/keyfile.json')
+    >>> fin = open('gs://bucket/key', transport_params=dict(client=client), ...)
+
+If you need more credential options, you can create an explicit ``google.auth.credentials.Credentials`` object
+and pass it to the Client.
+
+.. code-block:: python
+
+	>>> import os
+	>>> from google.auth.credentials import Credentials
```
Comment (mpenkov) [inline README.rst]: We want these examples to be self-contained.

```suggestion
	>>> from google.auth.credentials import Client, Credentials
```

**Comment by petedannemann** [comment]
@mpenkov thanks for the review. I addressed your comments

File: README.rst (hunk 1 above)
Comment (petedannemann) [reply README.rst]: I ended up using one of Google's [public datasets](https://cloud.google.com/storage/docs/public-datasets/landsat) instead

**Comment by petedannemann** [comment]
Sure thing, I just added that to the docs.

**Pull request merged**